### PR TITLE
HC alpha rebased

### PR DESCRIPTION
**Pull request by velzevur**
This is the latest released HC version. We would still be accumulating changes but those PRs better be opened against `master`.

The CI passes, now what is left is 

- [x] passing tests
- [x] a full sync of a node based on this PR
- [x] reviews :)

### CODE REVIEW

File: apps/aecore/src/aec_consensus_smart_contract.erl [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+        , get_type/0
+        , get_block_producer_configs/0
+        , is_leader_valid/3
+        ]).
+
+-include_lib("aecontract/include/hard_forks.hrl").
+-include("blocks.hrl").
+-include("aec_consensus.hrl").
+
+can_be_turned_off() -> false.
```
Comment (uwiger) [inline apps/aecore/src/aec_consensus_smart_contract.erl:77]: Is this correct?

File: apps/aecore/src/aec_parent_connector.erl [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+pmap(Fun, L, Timeout) ->
+    Workers =
+        lists:map(
+            fun(E) ->
+                spawn_monitor(
+                    fun() ->
+                        {WorkerPid, WorkerMRef} =
+                            spawn_monitor(
+                                fun() ->
```
Comment (uwiger) [inline apps/aecore/src/aec_parent_connector.erl:189]: Why the double spawn?

Comment (uwiger) [reply apps/aecore/src/aec_parent_connector.erl:189]: Guessing that it's because of timeouts?

File: apps/aecore/src/aec_preset_keys.erl [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+start_link(Keys) when is_map(Keys) ->
+    gen_server:start_link({local, ?SERVER}, ?MODULE, Keys, []).
+
+
+stop() ->
+    gen_server:stop(?SERVER).
+
+
+init(Keys) when is_map(Keys) ->
+    {ok, #state{ keys = Keys }}.
```
Comment (uwiger) [inline apps/aecore/src/aec_preset_keys.erl:116]: Shouldn't this process have `process_flag(sensitive, true)` given that it holds privkeys?

File: apps/aehttp/src/aehttpc_aeternity.erl [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    aetx_sign:new(Tx, Signatures).
+
+get_request(Path, Host, Port, Timeout) ->
+    try
+    Url = url(binary_to_list(Host), Port, false),
+    UrlPath = lists:concat([Url, binary_to_list(Path)]),
+    Req = {UrlPath, []},
+    HTTPOpt = [{timeout, Timeout}],
+    Opt = [],
+    lager:info("Req: ~p, with URL: ~ts", [Req, Url]),
```
Comment (uwiger) [inline apps/aehttp/src/aehttpc_aeternity.erl:135]: Shouldn't these be `lager:debug` messages in the long term?

File: apps/aehttp/src/aehttpc_aeternity.erl [hunk 5]
```diff
@@ -0,0 +1,166 @@
+-module(aehttpc_aeternity).
```
Comment (uwiger) [inline apps/aehttp/src/aehttpc_aeternity.erl:1]:
```suggestion
%%% @copyright (C) 2022, Aeternity
-module(aehttpc_aeternity).
```

File: apps/aehttp/src/aehttpc_btc.erl [hunk 6]
```diff
@@ -0,0 +1,181 @@
+-module(aehttpc_btc).
```
Comment (uwiger) [inline apps/aehttp/src/aehttpc_btc.erl:1]:
```suggestion
%%% @copyright (C) 2022, Aeternity
-module(aehttpc_btc).
```

File: apps/aehttp/test/aehttp_ae_sim.erl [hunk 7]
```diff
@@ -0,0 +1,204 @@
+%%%-----------------------------------------------------------------------------
+%%% @title /Users/sean/Aeternity/aeternity-hyperchains/apps/aehttp/test/aehttp_btc_sim
+%%% @doc ae node simulator embedded in minimum http api
+%%%
+%%% @author sean
+%%% @copyright <COPY_WRITE>
```
Comment (uwiger) [inline apps/aehttp/test/aehttp_ae_sim.erl:6]:
```suggestion
%%% @copyright (C) 2022, Aeternity
```

File: apps/aehttp/test/aehttp_parent_connector_tests.erl [hunk 8]
```diff
@@ -0,0 +1,119 @@
+-module(aehttp_parent_connector_tests).
```
Comment (uwiger) [inline apps/aehttp/test/aehttp_parent_connector_tests.erl:1]:
```suggestion
%%% @copyright (C) 2022, Aeternity
-module(aehttp_parent_connector_tests).
```

File: apps/aehttp/test/aehttp_btc_sim.erl [hunk 9]
```diff
@@ -0,0 +1,422 @@
+%%%-----------------------------------------------------------------------------
+%%% @title /Users/sean/Aeternity/aeternity-hyperchains/apps/aehttp/test/aehttp_btc_sim
+%%% @doc Extremely basic web server simulating a bitcoin node 
+%%% supporting a small number of hard coded bitcoin operations.
+%%% for testing hyperchains parent chain connector
+%%%
+%%% @author sean
+%%% @copyright <COPY_WRITE>
```
Comment (uwiger) [inline apps/aehttp/test/aehttp_btc_sim.erl:8]:
```suggestion
%%% @copyright (C) 2022, Aeternity```

File: data/aecore/.ceres/hc_smart_contract_accounts.json [hunk 10]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (uwiger) [inline data/aecore/.ceres/hc_smart_contract_accounts.json:1]: Should this be identical to `hc_smart_contrac_accounts_test.json`?

File: apps/aecore/src/aec_consensus_smart_contract.erl (hunk 1 above)
Comment (velzevur) [reply apps/aecore/src/aec_consensus_smart_contract.erl:77]: At this point - yes

File: apps/aecore/src/aec_preset_keys.erl (hunk 3 above)
Comment (velzevur) [reply apps/aecore/src/aec_preset_keys.erl:116]: We need a better solution for this, it had been tracked as #3937

File: apps/aecore/src/aec_parent_connector.erl (hunk 2 above)
Comment (velzevur) [reply apps/aecore/src/aec_parent_connector.erl:189]: This is my best guess as well

File: data/aecore/.ceres/hc_smart_contract_accounts.json (hunk 10 above)
Comment (velzevur) [reply data/aecore/.ceres/hc_smart_contract_accounts.json:1]: Yes. They should be generated according the network ID. This is already done in a different branch. Anyway, I tracked it in an issue #3974

**Review by uwiger — APPROVED** [review]

**Review by seanhinde — APPROVED** [review]
Don't see anything that should mess up normal operation. Onward to hyperchains in master!

**Pull request merged**